src/app/static/src/app/components/genericChart/GenericChart.vue

```diff
@@ -253,12 +295,39 @@
 
                     result[dataSourceId] = filterData(unfilteredData, filters, selectedFilterOptions);
                 }
-                return result;
+
+                if (result["data"]) {
```

Comment:
Yeah, that's because it was kind of a hack to get the subplots working. In theory, generic chart could support displaying multiple datasets - in practice this is in the only instance we have. Earlier incarnations of it didn't use subplot configuration, but once we needed that to add annotations etc into plotly, it was either maintain a dict of these things, or treat 'data' as a special case which can support subplots, and we went for the latter.

Reply:
Ok, I don't fully understand but I figured it was something like that!